sound/soc/sof/probe.c

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license.  When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Cezary Rojewski <cezary.rojewski@intel.com>
+//
+
+#include "sof-priv.h"
+#include "probe.h"
+
+int sof_ipc_probe_init(struct snd_sof_dev *sdev,
+		size_t buffer_size)
+{
+	struct sof_ipc_probe_dma_set_params *msg;
+	struct sof_ipc_reply reply;
+	unsigned int stream_tag = INVALID_NODE_ID;
+	size_t size;
+	int ret;
+
+	size = struct_size(msg, dma, 1);
+	msg = kzalloc(size, GFP_KERNEL);
+	if (!msg)
+		return -ENOMEM;
+
+	if (sdev->extractor)
+		stream_tag = sdev->extractor;
```

Comment:
You could remove the `stream_tag` initialisation and just do:
```
stream_tag = sdev->extractor ? : INVALID_NODE_ID;
```

Reply:
`stream_tag = sdev->extractor ? : INVALID_NODE_ID;`

I believe something missing here:
`stream_tag = sdev->extractor ? sdev->extractor : INVALID_NODE_ID;`
Honestly, that's just a matter of taste. Otped-out of tenary operator and went with if.

Reply:
Decided to aligned with your suggestion in v5.